### PR TITLE
Web Actions Tab: Basic Filtering

### PR DESCRIPTION
**Pull request by adrw**
Screen Shot 2019-03-08 at 6 30 41 PM

### CODE REVIEW

File: misk/web/tabs/web-actions/src/ducks/webActions.ts [hunk 1]
```diff
@@ -197,10 +209,11 @@ function* handleMetadata() {
           authFunctionAnnotations,
           dispatchMechanism: [action.dispatchMechanism],
           function: action.function.split("fun ").pop(),
+          hash: groupByWebActionHash(action),
           nonAccessOrTypeFunctionAnnotations
         }
       })
-      .groupBy(groupByWebActionHash)
+      .groupBy("hash")
```
Comment (wesleyk) [inline misk/web/tabs/web-actions/src/ducks/webActions.ts]: this is pretty magical, is there something more deterministic to use here?

File: misk/web/tabs/web-actions/src/containers/WebActionsContainer.tsx [hunk 2]
```diff
@@ -430,64 +444,91 @@ const SkeletonText = () => (
...
+          {...props}
+        />
       </div>
     )
   } else {
     // Displays mock of 5 Web Action cards which fill in when data is available
     return (
       <div>
-        <SkeletonWebActions />
+        <SkeletonWebActions {...props} />
```
Comment (wesleyk) [inline misk/web/tabs/web-actions/src/containers/WebActionsContainer.tsx]: why do the mocks need this props reference?

File: misk/web/tabs/web-actions/src/ducks/webActions.ts [hunk 3]
```diff
@@ -43,6 +44,15 @@ export interface IWebActionInternal {
   returnType: string
 }
 
+export const WebActionInternal: { [key: string]: string } = {
```
Comment (wesleyk) [inline misk/web/tabs/web-actions/src/ducks/webActions.ts]: small comment explaining what this is?

File: misk/web/tabs/web-actions/src/ducks/webActions.ts (hunk 1 above)
Comment (adrw) [reply misk/web/tabs/web-actions/src/ducks/webActions.ts]: The `groupByWebActionHash` function previously referenced now saves the value in a key in the object called `"hash"`. Maybe a different named key would seem less magical.

Comment (wesleyk) [reply misk/web/tabs/web-actions/src/ducks/webActions.ts]: we're relying on the fact that the key is "hash". Can we name it something more relevant and extract that to not a magic string?

File: misk/web/tabs/web-actions/src/containers/WebActionsContainer.tsx (hunk 2 above)
Comment (adrw) [reply misk/web/tabs/web-actions/src/containers/WebActionsContainer.tsx]: Initially had put in so the new filter form would be in the mock but realized I did it wrong so going to fix it and push.

**Pull request merged**